var/spack/repos/builtin/packages/py-pyzmq/package.py

```diff
@@ -60,8 +60,8 @@ class PyPyzmq(PythonPackage):
     # Undocumented dependencies
     depends_on("py-gevent", type=("build", "run"))
 
-    @run_before("install")
     @when("@15:19")
+    @run_before("install")
```

Comment:
Let's do
```suggestion
    @run_before("install", when="@15:19")
```
then, seems the least confusing.

Reply:
:+1: The when argument was explicitly added to fix issues with the order of decorators.

Reply:
You learn something new every day!